### PR TITLE
Bhu

### PR DESCRIPTION
**Pull request by bdogra**
WEI transactions and correct connection between web app and contract. Yet to be tested.

### CODE REVIEW

File: scripts/deploy.js [hunk 1]
```diff
@@ -11,10 +11,10 @@ async function main() {
   const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
   const unlockTime = currentTimestampInSeconds + ONE_YEAR_IN_SECS;
 
-  const lockedAmount = hre.ethers.utils.parseEther("1");
-
   const SlotMachine = await hre.ethers.getContractFactory("SlotMachine");
-  const slotMachine = await SlotMachine.deploy();
+  const slotMachine = await SlotMachine.deploy({
+    value: hre.ethers.utils.parseEther("0.1"),
```
Comment (pri358) [inline scripts/deploy.js:16]: what does this value map to in smart contract?

File: scripts/run.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const main = async () => {
     console.log("Contract deployed to:", slotMachineContract.address);
     console.log("Owner of the contract:", owner.address);
 
-    const randNums = await slotMachineContract.logicOfLuck("0.23");
+    const randNums = await slotMachineContract.play(web3.utils.toWei("0.1", 'ether'));
```
Comment (pri358) [inline scripts/run.js:20]: why is this converted to wei? If we are doing the conversion in the smart contract?

File: src/App.js [hunk 3]
```diff
@@ -91,24 +91,25 @@ const App = () => {
...
+      // const tempA = await symbolGenerator();
+      // const tempB = await symbolGenerator();
+      // const tempC = await symbolGenerator();
+
+      // setValA(tempA);
+      // setValB(tempB);
+      // setValC(tempC);
+      // setBetAmount('');
       
-      if(tempA===tempB && tempB===tempC)
```
Comment (pri358) [inline src/App.js:104]: why is this logic here? Aren't we doing this logic in smart contract?

File: scripts/deploy.js (hunk 1 above)
Comment (bdogra) [reply scripts/deploy.js:16]: This is essentially funding the contract with the required amount of ETH. In this case the value of the funding being 0.1.

File: scripts/run.js (hunk 2 above)
Comment (bdogra) [reply scripts/run.js:20]: The conversion happens in the web app (front end) using web3.js. For local testing (run.js) need to provide same input of WEI.

File: src/App.js (hunk 3 above)
Comment (bdogra) [reply src/App.js:104]: This logic is in smart contract, hence has been commented out (for the time being. Will be removed.) in this iteration.

File: scripts/deploy.js (hunk 1 above)
Comment (pri358) [reply scripts/deploy.js:16]: you mean the contract balance?

File: src/App.js (hunk 3 above)
Comment (pri358) [reply src/App.js:104]: ah my bad, i thought it was commented before and you un commented

File: contracts/SlotMachine.sol [hunk 4]
```diff
@@ -21,7 +22,7 @@ contract SlotMachine {
     function play(uint256 _betAmount) public payable {
         // Uncomment below code once dummy user has balance
 
-        uint userBalance = msg.value ;
+        uint256 userBalance = msg.value * (10**18);
```
Comment (pri358) [inline contracts/SlotMachine.sol:25]: this is the wei conversion no?

File: scripts/deploy.js (hunk 1 above)
Comment (bdogra) [reply scripts/deploy.js:16]: yes. If you see in the previous iteration:
lockedAmount = hre.ethers.utils.parseEther("1");
was unused and the contract had no balance.

Comment (pri358) [reply scripts/deploy.js:16]: Got it, I meant to ask this value=0.1 is the contract balance, correct?

File: contracts/SlotMachine.sol (hunk 4 above)
Comment (bdogra) [reply contracts/SlotMachine.sol:25]: This WEI conversion is for the user balance. The WEI conversion we do in front end is for the bet amount. 
Although, I am not sure is the user balance is returned as ETH or WEI. This still needs testing.

File: scripts/deploy.js (hunk 1 above)
Comment (bdogra) [reply scripts/deploy.js:16]: Yes

File: contracts/SlotMachine.sol (hunk 4 above)
Comment (pri358) [reply contracts/SlotMachine.sol:25]: Got it

**Pull request merged**